device-driver/examples/test-macro-driver.rs

```diff
@@ -178,6 +178,44 @@ pub mod registers {
                 /// Baudrate value
                 value: u16 = 0..16,
             },
+            /// Instance of the Foo block
+            block Foo0 {
+                const BASE_ADDRESS: u8 = 64;
+
+                /// Baudrate register
+                register Baudrate {
+                    type RWType = RW;
+                    type ByteOrder = LE; // Everything is BE by default
+                    const ADDRESS: u8 = 0;
+                    const SIZE_BITS: usize = 16;
+
+                    /// Baudrate value
+                    value: u16 = 0..16,
+                },
+            },
+            /// Second instance of the Foo block
+            ref Foo1 = Foo0 {
+                const BASE_ADDRESS: u8 = 80;
+            },
```

Comment:
Interesting.

At some point I might want to expand `copy_of` to single registers (and not full blocks) and who knows what else...

I'm thinking whether this syntax will get in the way at some point.
For example, when `ref`ing a register instead of a block, we'd probably want to have an `ADDRESS` const instead of `BASE_ADDRESS`.
Not having the name of the 'type' we're ref-ing will complicate things.

So just for forward compatibility, we should make it `ref block Foo1` or `block ref Foo1`. Which one would you prefer? I think I like `ref block` best. It sets precedence for having other keywords in front, e.g. `proto block` which could be just the definition of a block without instantiation so that you can give the block types nice names. Idk... Not sure if that will ever be a thing.

Sorry for the rambling :P

Reply:
I agree that `copy_of` should eventually be expanded to single registers, other refs, etc. But I don't think we should make the contents of a ref dependent on the type of thing being copied. Perhaps the syntax should be something like:
`ref Foo1: Foo0 = 80;`
or
`ref Foo1 = Foo0 @ 80;`

Reply:
I see what you're thinking. However, that's assuming that the only thing a copy_of ever changes is the address. At least for a register, one might want to change the readwrite permissions. This happens a lot where you have interrupt registers with status, set, clear, etc.

So while a bit more verbose, I like my suggestion better because it leaves more options open for the future

Reply:
One thing to note here is that the generation already supports copying registers as well as blocks (though not overriding any of the register's properties). So right now you could do:

```
register Foo { ... },
ref Bar = Foo { const BASE_ADDRESS: u8 = 12; }
```

And it will compile. Adding the `block` keyword won't actually change anything in the generation so `ref block Bar = Foo ...` will still compile, which seems wrong. If you want to allow overriding properties in a "copy_of", that may require re-thinking how it is modeled and how generation works.

Reply:
Never mind, I think I have a solution for the above. It will just take a bit of reworking.